test/scripts/dynamic-fee-teal-test.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+date '+dynamic-fee-teal-test start %Y%m%d_%H%M%S'
+
+set -e
+set -x
+set -o pipefail
+export GOPATH=$(go env GOPATH)
+
+TEMPDIR=$(mktemp -d)
+trap "rm -rf $TEMPDIR" 0
+
+NETDIR=${TEMPDIR}/net
+
+if [ ! -z $BINDIR ]; then
+    export PATH=${BINDIR}:${PATH}
+fi
+
+goal network create -r ${NETDIR} -n dynamic-fee-teal-test -t ${GOPATH}/src/github.com/algorand/go-algorand/test/testdata/nettemplates/TwoNodes50EachFuture.json
+
+goal network start -r ${NETDIR}
+
+# replaces prior trap0
+trap "goal network stop -r ${NETDIR}; rm -rf ${TEMPDIR}" 0
+
+export ALGORAND_DATA=${NETDIR}/Node
+
+ACCOUNT=$(goal account list|awk '{ print $3 }')
+ACCOUNTB=$(goal account new|awk '{ print $6 }')
+ACCOUNTC=$(goal account new|awk '{ print $6 }')
+ACCOUNTD=$(goal account new|awk '{ print $6 }')
+ZERO_ADDRESS=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY5HFKQ
+LEASE=uImiLf+mqOqs0BFsqIUHBh436N/z964X50e3P9Ii4ac=
+
+# Fund ACCOUNTB
+goal clerk send -a 100000000 -f ${ACCOUNT} -t ${ACCOUNTB}
+
+# Generate the template
+algotmpl -d ${GOPATH}/src/github.com/algorand/go-algorand/tools/teal/templates/ dynamic-fee --amt=1000000 --cls=${ACCOUNTC} --to=${ACCOUNTD} --fv=1 --lv=1001 --lease=${LEASE} > ${TEMPDIR}/dynamic.teal
+
+#
+# Fee will come from ACCOUNT in the first transaction
+# We will close out to ACCOUNTC in second transaction
+# We will pay ACCOUNTD with 1000000 microAlgos in second transaction
+#
+
+# Compile the template
+goal clerk compile -a ${ACCOUNTB} -s ${TEMPDIR}/dynamic.teal -o ${TEMPDIR}/dynamic.steal
```

Comment:
steal is used for Derek's lang if I remember correctly

Reply:
changed